src/directives/style-map.ts

```diff
@@ -36,7 +40,18 @@ class StyleDeclaration {
              if (v === null) {
                return value;
              }
+             // Convert property names from camel-case to dash-case, i.e.:
+             //  `backgroundColor` -> `background-color`
+             // Vendor-prefixed names need an extra `-` appended to front:
+             //  `webkitAppearance` -> `-webkit-appearance`
+             // Exception is any property name containing a dash, including
+             // custom properties; we assume these are already dash-cased i.e.:
+             //  `--my-button-color` --> `--my-button-color`
+             // Note that prefixed
              if (prop.indexOf('-') === -1) {
+               if (VENDOR_PREFIX.test(prop)) {
+                 prop = '-' + prop;
```

Comment:
Wouldn't this incorrectly convert `--webkit-prop` to `---webkit-prop`?

Reply:
No, because it looks for vendor-prefix'ed _camel case_ properties (prefix immediately followed by uppercase `[A-Z]`).

What you pass into `styleMap` are `style.<prop>`-assignable property names (except for custom properties, which are passed as dash-case and set via `setProperty`). 

`style.webkitAppearance` is a thing, so this assumes you'd want to pass `webkitAppearance` as a styleMap key.

Reply:
I will add a `^` to the regex to only match prefixes at the start, though.

Reply:
Won't match custom properties this only tests properties without any dashes.